src/Symfony/Component/Validator/Constraints/ExpressionValidator.php

```diff
@@ -46,6 +47,10 @@ public function validate($value, Constraint $constraint)
         $variables['value'] = $value;
         $variables['this'] = $this->context->getObject();
 
+        if (strlen($constraint->dataPath) > 0) {
```

Comment:
Hm i don't like this because than it can be a `string` or instanceof `PropertyPath`. If we keep in mind that we want to use later on scalar type hints, we would have a problem.